primitives/arithmetic/src/traits.rs

```diff
@@ -527,14 +531,14 @@ mod ensure {
 		/// assert_eq!(overflow(), Err(ArithmeticError::Overflow));
 		/// ```
 		fn ensure_sub_assign(&mut self, v: Self) -> Result<(), ArithmeticError> {
-			*self = self.ensure_sub(v)?;
+			self.checked_sub(&v).ok_or_else(|| error::inverse(&v))?;
```

Comment:
```suggestion
			*self = self.checked_sub(&v).ok_or_else(|| error::inverse(&v))?;
```

Reply:
No tests, just seeing that now :scream: - will fix

Reply:
Yes, there are no happy path tests 😅